syntax="proto2";
import "nanopb.proto";
package rina.messages;
option optimize_for = LITE_RUNTIME;

message objectValue_t {
	// value of an object
	
	optional int32	intval = 1;
	optional sint32	sintval = 2;
	optional int64	int64val = 3;
	optional sint64	sint64val = 4;
	optional string	strval = 5 [(nanopb).max_size = 10];
	optional bytes	byteval = 6 [(nanopb).max_size = 256];			// arbitrary structure or message
	optional fixed32 floatval = 7;
	optional fixed64 doubleval = 8;
}

message nameValuePair_t {								//A name-value pair
	optional string name = 1 [(nanopb).max_size = 40];
	optional objectValue_t value = 2;
}

message property_t {         // carries the information of a name/value pair
        required string name = 1 [(nanopb).max_size = 40];
        required string value = 2 [(nanopb).max_size = 40];
}

message a_data_t {
      optional int32 sourceAddress = 1;
      optional int32 destAddress = 2;
      optional bytes cdapMessage = 3;
}
